### PR TITLE
samples: common: improvements in net_connect()

### PR DESCRIPTION
**Pull request by mniestroj**
Those improvements affect mostly `esp32` and `mimxrt1060_evkb` platforms, which require waiting for specific condition on network interface, before it can be used by `net_connect()` implementation.

### CODE REVIEW

File: samples/common/net_connect.c [hunk 1]
```diff
@@ -12,10 +12,65 @@ LOG_MODULE_REGISTER(golioth_samples, LOG_LEVEL_DBG);
...
+	} else {
+		wait_for_iface_up_poll(iface);
+	}
+}
+
 void net_connect(void)
 {
 	struct net_if *iface = net_if_get_default();
 
+	wait_for_iface_up(iface);
```
Comment (ncmiller) [inline samples/common/net_connect.c:89]: Might be helpful to put a log message here, in case it gets stuck.

**Review by ncmiller — APPROVED** [review]
Thanks for the quick fixes!

File: samples/common/net_connect.c (hunk 1 above)
Comment (mniestroj) [reply samples/common/net_connect.c:90]: Added `LOG_INF("Waiting for interface to be up");`. Is that fine?

Comment (ncmiller) [reply samples/common/net_connect.c:90]: Perfect, thanks

**Review by szczys — APPROVED** [review]
LGTM!

**Pull request merged**